File: vane-velocity/src/main/java/org/oddlama/velocity/compat/VelocityCompatProxyPlayer.java
package org.oddlama.velocity.compat;

import com.velocitypowered.api.proxy.Player;
import java.util.UUID;
import net.kyori.adventure.text.Component;
import org.oddlama.vane.proxycore.ProxyPlayer;

public class VelocityCompatProxyPlayer implements ProxyPlayer {

    final Player player;

    public VelocityCompatProxyPlayer(Player player) {
        this.player = player;
    }

    @Override
    public void disconnect(String message) {
        player.disconnect(Component.text(message));
    }

    @Override
    public UUID getUniqueId() {
        return player.getUniqueId();
    }

    @Override
    public long getPing() {
        return player.getPing();
    }

    @Override
    public void sendMessage(String message) {
        player.sendMessage(Component.text(message));
    }
}

File: vane-velocity/src/main/kotlin/org/oddlama/velocity/compat/VelocityCompatProxyPlayer.kt
package org.oddlama.velocity.compat

import com.velocitypowered.api.proxy.Player
import net.kyori.adventure.text.Component
import org.oddlama.vane.proxycore.ProxyPlayer
import java.util.UUID

class VelocityCompatProxyPlayer(val player: Player) : ProxyPlayer {
    override fun disconnect(message: String?) {
        player.disconnect(Component.text(message.orEmpty()))
    }

    override val uniqueId: UUID?
        get() = player.uniqueId

    override val ping: Long
        get() = player.ping

    override fun sendMessage(message: String?) {
        player.sendMessage(Component.text(message.orEmpty()))
    }
}
